azure/services/resourceskus/client.go

```diff
@@ -19,55 +19,53 @@ package resourceskus
 import (
 	"context"
 
-	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-11-01/compute"
-	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v5"
 	"github.com/pkg/errors"
 	"sigs.k8s.io/cluster-api-provider-azure/azure"
 	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
 )
 
 // Client wraps go-sdk.
 type Client interface {
-	List(context.Context, string) ([]compute.ResourceSku, error)
+	List(context.Context, string) ([]armcompute.ResourceSKU, error)
 }
 
 // AzureClient contains the Azure go-sdk Client.
 type AzureClient struct {
-	skus compute.ResourceSkusClient
+	skus armcompute.ResourceSKUsClient
```

Comment:
Should we make this a pointer for consistency?
```suggestion
	skus *armcompute.ResourceSKUsClient
```

Reply:
Yes indeed, I must have missed this one.